tests/unit/test_commands.py

```diff
@@ -120,3 +122,35 @@ def is_requirement_command(command: Command) -> bool:
         return isinstance(command, RequirementCommand)
 
     check_commands(is_requirement_command, ["download", "install", "wheel"])
+
+
+@pytest.mark.parametrize(
+    "command", ["cache", "check", "config", "freeze", "hash", "help", "inspect", "show"]
```

Comment:
Instead of this, consider using `set(commands_dict.keys()).symmetric_difference({some, specific, commands})`? That way, only the commands which need network support would need to be listed (i.e. opt-in) and it'll protect against typos in this listing as well.

Reply:
I like this. This has been implemented in https://github.com/pypa/pip/pull/12566/commits/35ccbd1da5e3f294726389df497fd466cb3ecccb.